src/mongocxx/collection.cpp

```diff
@@ -185,6 +186,10 @@ cursor collection::find(bsoncxx::document::view filter, const options::find& opt
         filter_builder << "$orderby" << bsoncxx::types::b_document{*options.sort()};
     }
 
+    if (options.max_time()) {
+        filter_builder << "$maxTimeMS" << (*options.max_time()).count();
```

Comment:
optional will let you call methods on the underlying object directly with -> so:
```
options.max_time()->count();
```